### PR TITLE
Add new timer module for performance.now shim

### PR DESCRIPTION
**Pull request by cannoneyed**
Fixes https://github.com/magenta/magenta-js/issues/471

Adds a shim module for `window.performance` for Node.js use that's only included during the Node webpack build.

### CODE REVIEW

**Comment by cannoneyed** [comment]
Yeah, I'd be happy to add a `fetch` shim as well... on a side note it's really annoying that this is even a thing.

**Comment by notwaldorf** [comment]
Yeah it’s really not great. I think you can just use that dependency I use in the README, and maybe have two different files like you do for performance. Sigh.

**Comment by cannoneyed** [comment]
@notwaldorf would it be better to put the `performance`/`fetch` PRs into one? I have a branch that fixes all the fetch stuff, happy to pull it all together into one file or one PR if it makes things easier...

**Comment by notwaldorf** [comment]
Whatever you prefer! Since you have a separate branch, let's do it as separate PRs, so I'll merge this one!

File: music/src/core/timer.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * A module containing a metronome based on Tone.js. The timing is done
+ * using the underlying WebAudio clock, so it is accurate, and the metronome
+ * fires callbacks for every audible click, quarter and bar marks.
+ *
+ * @license
+ * Copyright 2012 Google Inc. All Rights Reserved.
```
Comment (notwaldorf) [inline music/src/core/timer.ts:7]: 2020!

Comment (notwaldorf) [inline music/src/core/timer.ts:7]:
```suggestion
 * Copyright 2020 Google Inc. All Rights Reserved.
```

File: music/src/core/timer_node.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**
+ * A module containing a metronome based on Tone.js. The timing is done
+ * using the underlying WebAudio clock, so it is accurate, and the metronome
+ * fires callbacks for every audible click, quarter and bar marks.
+ *
+ * @license
+ * Copyright 2012 Google Inc. All Rights Reserved.
```
Comment (notwaldorf) [inline music/src/core/timer_node.ts:7]:
```suggestion
 * Copyright 2020 Google Inc. All Rights Reserved.
```

File: music/demos/coconet.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ import * as tf from '@tensorflow/tfjs-core';
 import {mergeConsecutiveNotes, replaceInstruments} from '../src/core/sequences';
 import * as mm from '../src/index';
 import {NoteSequence} from '../src/index';
+import * as timer from '../src/timer';
```
Comment (notwaldorf) [inline music/demos/coconet.ts:23]: why isn't this src/core/timer?

Comment (notwaldorf) [reply music/demos/coconet.ts:23]: (yeah, i think this is a typo)

Comment (cannoneyed) [reply music/demos/coconet.ts:23]: THX, that's a mistake

**Pull request merged**